File: removeCharacters.cpp
/*
#2 Remove Character
Given two strings string1 and string2, remove those characters from first string(string1) 
which are present in second string(string2).
Topic: Array
Level: Easy
*/

#include <iostream>
#include <string>
using namespace std;

string removeCommonCharacters(string str1, string str2)
{
    for (int i = 0; i < str1.length(); i++)
    {
        for (int j = 0; j < str2.length(); j++)
        {
            if (str1[i] == str2[j])
            {
                str1.erase(i, 1);
            }
        }
    }
    return str1;
}

int main()
{
    cout << "Enter your first string: ";
    string str1;
    getline(cin, str1);

    cout << "Enter your second string: ";
    string str2;
    getline(cin, str2);

    
    cout << "After removing common characters: " << removeCommonCharacters(str1, str2);
}

File: remove_characters.py
'''
#2 Remove Character
Given two strings string1 and string2, remove those characters from first string(string1) 
which are present in second string(string2).
Topic: Array
Level: Easy
'''


# Python3 program to remove common characters from two strings
def remove_common_characters(str1, str2):
    result = ""

    for char in str1:
        if char not in str2:
            result += char
    return result 

# def remove_common_characters(string1, string2):
#     return "".join([char for char in string1 if char not in string2])

str1 = input("Enter the first string: ")
str2 = input("Enter the second string: ")
print("String after removing characters:", remove_common_characters(str1, str2))
